edb/pgsql/delta.py

```diff
@@ -4850,59 +4864,79 @@ def _compile_conversion_expr(
+        # ref_tables = schemamech.get_ref_storage_info(ir.schema, refs)
+        # local_table_only = all(
+        #     t == old_ptr_stor_info.table_name
+        #     for t in ref_tables
+        # )
 
         ptr_path_id = tgt_path_id.ptr_path()
         src_path_id = ptr_path_id.src_path()
         assert src_path_id
 
-        external_rels = {}
-        external_rvars = {}
+        src_rel = compiler.new_external_rel(
+            rel_name=(source_alias,),
+            path_id=src_path_id,
+        )
+        src_rel.path_outputs[(src_path_id, 'iterator')] = \
+            pgast.ColumnRef(name=('id',))
+        external_rels = {
+            src_path_id: (src_rel, ('value', 'source', 'identity', 'iterator'))
+        }
 
-        if produce_ctes:
-            external_rels[src_path_id] = compiler.new_external_rel(
-                rel_name=(source_alias,),
-                path_id=src_path_id,
-            ), ('value', 'source')
-        else:
+        if for_each_pointer:
             if ptr_table:
-                rvar = compiler.new_external_rvar(
-                    rel_name=(source_alias,),
-                    path_id=ptr_path_id,
-                    outputs={
-                        (src_path_id, ('identity',)): 'source',
-                    },
-                )
-                external_rvars[ptr_path_id, 'source'] = rvar
-                external_rvars[ptr_path_id, 'value'] = rvar
-                external_rvars[src_path_id, 'identity'] = rvar
-                external_rvars[tgt_path_id, 'identity'] = rvar
-                if local_table_only and not is_lprop:
-                    external_rvars[src_path_id, 'source'] = rvar
-                    external_rvars[src_path_id, 'value'] = rvar
-                elif is_lprop:
-                    external_rvars[tgt_path_id, 'value'] = rvar
+                src_rel.path_outputs[(ptr_path_id, 'identity')] = \
+                    pgast.ColumnRef(name=('ctid',))
+                src_rel.path_outputs[(tgt_path_id, 'identity')] = \
+                    pgast.ColumnRef(name=('ctid',))
+                src_rel.path_outputs[(tgt_path_id, 'iterator')] = \
+                    pgast.ColumnRef(name=('ctid',))
             else:
-                src_rvar = compiler.new_external_rvar(
-                    rel_name=(source_alias,),
-                    path_id=src_path_id,
-                    outputs={},
-                )
-                external_rvars[src_path_id, 'identity'] = src_rvar
-                external_rvars[src_path_id, 'value'] = src_rvar
-                external_rvars[src_path_id, 'source'] = src_rvar
+                src_rel.path_outputs[(tgt_path_id, 'iterator')] = \
+                    pgast.ColumnRef(name=('id',))
+
+            external_rels[ptr_path_id] = \
+                (src_rel, ('source', 'value'))
+            external_rels[tgt_path_id] = \
+                (src_rel, ('identity', 'source', 'value', 'iterator'))
+
+        #     if ptr_table:
```

Comment:
So I have the following DDL:
```
alter type Base {
    alter link foo {
        alter property bar set type int32;
    };
};
```

I set `external_rels` for `Base`, `Base::foo` and `Base::foo@`.
But on lookup for set with path_id `(default::Base).>foo[IS default::Child]@bar[IS std::int64]`, I'm not finding it in `external_rels`.

This should happen here: https://github.com/edgedb/edgedb/blob/7a145a9596fd21c4d5a330839459947e0582e79a/edb/pgsql/compiler/relgen.py#L426-L428

Reply:
Huh, you mean `external_rels` is getting cleared somewhere?